src/esn.calendar.libs/app/components/event/form/event-form.directive.js

```diff
@@ -19,6 +19,7 @@ function calEventForm() {
   ////////////
 
   function link(scope) {
+    scope.dragModal();
```

Comment:
- move the dragModal contents here. No need to polute $scope with a setup function
- use the second attribute of the link function (more doc at [Creating a Directive that Manipulates the DOM](https://docs.angularjs.org/guide/directive]) chapter) to narrow down your lookup of the element having drag-modal class. You can try to use [children](https://api.jquery.com/children/), that may work according to your DOM tree.

Reply:
Huy suggests that I can narrow down the finding and also call the function right inside controller like this, and that's how I'm doing at the moment. By doing this, I have tested and we also don't need `setTimeOut` anymore
 ``` 
scope.dragModal();

function dragModal() {
    $element.find('.modal-dialog').draggable({
      handle: '.modal-header, .modal-body, .modal-footer'
    });